### PR TITLE
Use new SmallRye and add Test for multiple ConfigSources

### PR DESCRIPTION
**Pull request by tevans78**
Signed-off-by: tevans <tevans@uk.ibm.com>

### CODE REVIEW

**Comment by tevans78** [comment]
#build

File: dev/io.openliberty.microprofile.config.2.0.internal_fat/fat/src/io/openliberty/microprofile/config/internal_fat/apps/defaultSources/DefaultSourcesTestServlet.java [hunk 1]
```diff
@@ -29,6 +33,33 @@
 @WebServlet("/")
 public class DefaultSourcesTestServlet extends FATServlet {
 
+    @Inject
```
Comment (Joseph-Cass) [inline dev/io.openliberty.microprofile.config.2.0.internal_fat/fat/src/io/openliberty/microprofile/config/internal_fat/apps/defaultSources/DefaultSourcesTestServlet.java:36]: This file needs a copyright header update

File: dev/io.openliberty.io.smallrye.config/src/io/openliberty/microprofile/config/internal/extension/OLSmallRyeConfigExtension.java [hunk 2]
```diff
@@ -10,15 +10,21 @@
...
 import io.smallrye.config.inject.ConfigExtension;
 
-@Component(service = WebSphereCDIExtension.class, configurationPolicy = ConfigurationPolicy.IGNORE, property = { "service.vendor=IBM" }, immediate = true)
-public class OLSmallRyeConfigExtension extends ConfigExtension implements Extension, WebSphereCDIExtension {
+@Component(service = CDIExtensionMetadata.class, configurationPolicy = ConfigurationPolicy.IGNORE, property = { "service.vendor=IBM" }, immediate = true)
+public class OLSmallRyeConfigExtension implements CDIExtensionMetadata {
+    @Override
+    public Set<Class<? extends Extension>> getExtensions() {
+        return Collections.singleton(ConfigExtension.class);
+    }
```
Comment (Joseph-Cass) [inline dev/io.openliberty.io.smallrye.config/src/io/openliberty/microprofile/config/internal/extension/OLSmallRyeConfigExtension.java:29]: Not entirely sure what this is doing, or why it's in this PR?

**Review by Joseph-Cass — CHANGES_REQUESTED** [review]
LGTM, other than the copyright header.

I'm also unsure why the `OLSmallRyeConfigExtension` class change is needed.

File: dev/io.openliberty.io.smallrye.config/src/io/openliberty/microprofile/config/internal/extension/OLSmallRyeConfigExtension.java (hunk 2 above)
Comment (tevans78) [reply dev/io.openliberty.io.smallrye.config/src/io/openliberty/microprofile/config/internal/extension/OLSmallRyeConfigExtension.java:29]: As part of the update to 2.2.0 I tried to remove as many build warnings as I could ... in this case it was the use of a deprecated interface.

**Comment by tevans78** [comment]
#build

File: dev/io.openliberty.io.smallrye.config/src/io/openliberty/microprofile/config/internal/extension/OLSmallRyeConfigBuilder.java [hunk 3]
```diff
@@ -29,6 +29,14 @@
 
     private static final TraceComponent tc = Tr.register(OLSmallRyeConfigBuilder.class);
 
+    public OLSmallRyeConfigBuilder() {
+        super();
+        //I have no idea why this property would default to true in the SmallRye code
+        //With it set to true, you get a validation error on hundreds of properties
+        //At the moment, I don't really understand what it is for
+        withValidateUnknown(false);
```
Comment (tevans78) [inline dev/io.openliberty.io.smallrye.config/src/io/openliberty/microprofile/config/internal/extension/OLSmallRyeConfigBuilder.java]: Should be able to remove this line again if this issue is resolved in a future version:
https://github.com/smallrye/smallrye-config/issues/548

**Review by Joseph-Cass — APPROVED** [review]

**Comment by Joseph-Cass** [comment]
#build
#spawn.fullfat.buckets=io.openliberty.microprofile.config.2.0.internal_fat_tck,com.ibm.ws.microprofile.config.1.4_fat,com.ibm.ws.microprofile.config.1.3_fat,com.ibm.ws.microprofile.config.1.2_fat,com.ibm.ws.microprofile.config.1.1_fat

**Comment by Joseph-Cass** [comment]
The PB failed `Feature verification` checks with errors such as:
```
ERROR: com.ibm.websphere.appserver.mpConfig-2.0
[BAD_PACKAGE_VERSION io.smallrye.config@2.3.0]
Changes were detected in package io.smallrye.config@2.3.0 which mean the package version is now incorrect. The baseline knows the package at version 2.0.0 Change the version of the package from its current version 2.3.0 to 3.0.0 Remember!! Major Changes in CD Open stream require POC Approval. If you do not have this, do not proceed with the changes that lead to this error.
For More Information, visit http://was.pok.ibm.com/xwiki/bin/view/Build/FeatureChecker
```

I've update the bundle version. Trying another PB:

#build
#spawn.fullfat.buckets=io.openliberty.microprofile.config.2.0.internal_fat_tck,com.ibm.ws.microprofile.config.1.4_fat,com.ibm.ws.microprofile.config.1.3_fat,com.ibm.ws.microprofile.config.1.2_fat,com.ibm.ws.microprofile.config.1.1_fat

**Comment by Joseph-Cass** [comment]
Wondering whether changing `bVersion` from 1.0 -> 2.0 caused the following:
```
ERROR: com.ibm.websphere.appserver.mpConfig-2.0
[MISSING_DEV_RESOURCE io.smallrye.config.ConfigMappings.class]
Path  io.smallrye.config.ConfigMappings.class is known to have changed for com.ibm.websphere.appserver.mpConfig-2.0 but there is no associated dev resource. This error should not be ignored unless this is an issue with a legacy resource that is part of an API/SPI package but is not exposed as API/SPI.
For More Information, visit http://was.pok.ibm.com/xwiki/bin/view/Build/FeatureChecker
```

Reverted the change and trying again.

#build
#spawn.fullfat.buckets=io.openliberty.microprofile.config.2.0.internal_fat_tck,com.ibm.ws.microprofile.config.1.4_fat,com.ibm.ws.microprofile.config.1.3_fat,com.ibm.ws.microprofile.config.1.2_fat,com.ibm.ws.microprofile.config.1.1_fat

**Comment by Joseph-Cass** [comment]
The build failed with the same issue in https://github.com/OpenLiberty/open-liberty/pull/16277#issuecomment-843050737

**Comment by Joseph-Cass** [comment]
I have made a WS-CD-Open PR to ignore the feature verification errors: https://github.ibm.com/websphere/WS-CD-Open/pull/24027

**Comment by Joseph-Cass** [comment]
Another PB needs running once a continuous build has pushed https://github.ibm.com/websphere/WS-CD-Open/pull/24027 to release.

**Pull request closed without merging**